torch/csrc/distributed/rpc/python_functions.cpp

```diff
@@ -41,6 +45,20 @@ std::shared_ptr<FutureMessage> py_rpc_builtin(
         Stack stack = torch::jit::createStackForSchema(
             op->schema(), args, kwargs, c10::nullopt);
 
+        auto& autogradContainer = DistAutogradContainer::getInstance();
+        if (autogradContainer.hasValidContext()) {
```

Comment:
Hmm. So if there is no current valid context, we just silently don't attach any autograd edges. That doesn't sound right to me: if there is no valid context, we had better *error* if we try to backpropagate.

Basically, if I don't do a context, and then I attend to rpc and backward through it, what error message do I get? That's something you should be testing for.